### ISSUE TITLE
Text or images in a pdf are loosing resolution, displaying in low quality

### ISSUE THREAD

**Issue by Shristi21** [mozilla/pdf.js#17599]
Attach (recommended) or Link to PDF file here:  
[AmadeusONE_Internal release notes_May-Dec23_Final 2.pdf](https://github.com/mozilla/pdf.js/files/14097167/AmadeusONE_Internal.release.notes_May-Dec23_Final.2.pdf)

Configuration:
- Web browser and its version:   Chrome and Version 120.0.6099.234
- Operating system and its version:  macOs and 14.2.1 (23C71) 
- PDF.js version:  Comparison between latest and 2.7 version
- Is a browser extension: No

Steps to reproduce the problem:
1.  Open the attached pdf in latest version of PDF.js 
2. You will notice the images in pdf are displaying in low quality seems like they are loosing the resolution.  
3. Open the same attached pdf in 2.7 version of PDF.js then there images are displaying in better quality. 

What is the expected behavior? (add screenshot).  Pdf's text and images should display in good quality

What went wrong? (add screenshot) Pdf's text/images are loosing resolution, displaying in low quality. 

PFA screenshot of comparison with both the versions -- 
<img width="1915" alt="Pdf js" src="https://github.com/mozilla/pdf.js/assets/66913303/2a77e38f-cbba-4211-9f6c-e7cc83a14e61">

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by Snuffleupagus** [comment]
Duplicate of #16273

**Comment by Shristi21** [comment]
I checked the issue number -- #16273  there it's mentioned that pdfjs is primarily for Firefox browser so, I tried opening the attached pdf in Firefox and there as well the quality of the pdf is low. 

PFA screenshot of the comparison of the attached pdf on Firefox with latest version of Pdf js and on chrome with 2.7 version of Pdf js.  Left one is on Firefox and right one is on chrome with 2.7 version of Pdf js. 
On 2.7 version of Pdf js, the quality of the attached pdf is good as compared to one in latest version on Firefox. 

<img width="1718" alt="firefox and google pdfjs" src="https://github.com/mozilla/pdf.js/assets/66913303/5b438592-842d-41f3-9002-25f79afc9357">

**Comment by Shristi21** [comment]
Hi @Snuffleupagus,

I have attached my findings on Pdf js for mozilla also , as you can see there is difference in the quality with latest version .

**Comment by calixteman** [comment]
That's right: it used to look better in Firefox.
I used [mozregression](https://mozilla.github.io/mozregression/) and I found that the patch in [bug 1730725](https://bugzilla.mozilla.org/show_bug.cgi?id=1730725) causes this "regression".
The real "culprit" is:
https://github.com/mozilla/pdf.js/pull/13991
Some of the images in the pdf have the flag `Interpolate` set to false and the above patch just helps to disable interpolation when it's required.
So the best way to fix your issue is just to set `Interpolate` to true in your pdf.
